### DISCUSSION TITLE
TestClient: how to test overwrites in the Config of an input (pydantic) model?

### DISCUSSION THREAD

**Discussion by camold** [fastapi/fastapi#6337] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
###### fastapi_extra.py
import fastapi
from pydantic import BaseModel


class MyModel(BaseModel):
    i: int

    class Config:
        extra = "ignore"


def create_app():

    app = fastapi.FastAPI()

    @app.post("/")
    def test(m: MyModel):
        return [m.i, MyModel.Config.extra]

    return app

########### test_fastapi_extra.py
import pytest
from fastapi.testclient import TestClient

import fastapi_extra

# overwrite from "ignore" to "forbid"
fastapi_extra.MyModel.Config.extra = "forbid"


@pytest.fixture
def test_app():
    yield TestClient(fastapi_extra.create_app())


def test_root(test_app):
    # send a message with an additional attribute
    r = test_app.post("/", json={"i": 2, "foo": "bar"})
    assert r.json() == [2, "forbid"]
    assert r.status_code == 200  # this should be 422 then!
```

### Description

I would like to ensure that the models/queries used in my pytests are up to date with the code. Hence, the goal was to set the `Config.extra` parameter to `"forbid"` in the pytests.

Note that the original application requires `"ignore"` to not break production. So `"forbid"` should only be used within the tests.

`pytest test_fastapi_extra.py`
should raise an error but does not... so weirdly enough the model config is set to "forbid" but the extra fields do not cause an unprocessable entity error.

### Operating System

Linux

### FastAPI Version

0.74.1

### Python Version

3.9.7

### Additional Context

pydantic 1.9.0
pytest 7.0.1

**Comment by camold** [comment]
If ever someone comes accross this problem. The solution was to use the values provided by `pydantic.Extra` instead of the strings for `"forbid"` and `"ignore"`. This works as expected.
Unfortunately the documentation in pydantic mentions that you could use either the string or the enum but apparently they are not working the same way.

```python
###### fastapi_extra.py
import fastapi
from pydantic import BaseModel, Extra


class MyModel(BaseModel):
    i: int

    class Config:
        extra = Extra.ignore # use the enum provided by pydantic here instead of "ignore" string


def create_app():

    app = fastapi.FastAPI()

    @app.post("/")
    def test(m: MyModel):
        return [m.i, MyModel.Config.extra]

    return app

########### test_fastapi_extra.py
import pytest
from fastapi.testclient import TestClient
from pydantic import Extra

import fastapi_extra

# overwrite from Extra.ignore to Extra.forbid using the enums again!
fastapi_extra.MyModel.Config.extra = Extra.forbid


@pytest.fixture
def test_app():
    yield TestClient(fastapi_extra.create_app())


def test_root(test_app):
    # send a message with an additional attribute
    r = test_app.post("/", json={"i": 2, "foo": "bar"})
    assert r.status_code == 422
    assert r.json() == {
        "detail": [
            {
                "loc": ["body", "foo"],
                "msg": "extra fields not permitted",
                "type": "value_error.extra",
            }
        ],
    }

```

**Comment by tiangolo** [comment, marked as answer]
Thanks for reporting back and closing the issue 👍
